components/MarkdownTextBlock/src/MarkdownTextBlock.xaml.cs

```diff
@@ -116,6 +114,7 @@ private void ApplyText(bool rerender)
             if (!string.IsNullOrEmpty(Text))
             {
                 this.MarkdownDocument = Markdown.Parse(Text, _pipeline);
+                OnMarkdownParsed?.Invoke(this, new MarkdownParsedEventArgs(this.MarkdownDocument));
```

Comment:
I believe there is a 1-1 relationship between MarkdownTextBlock and its MarkdownDocument, so I would remove the Document property from the `MarkdownParsedEventArgs`, as you can get that from the sender.

Reply:
Every time you change the `Text` property of MarkdownTextBlock, technically you would get a brand new MarkdownDocument tree. If we also move the parsing to another thread, technically there would be no guarantees that the MarkdownDocument currently set on the control is the one corresponding to the given event, especially in streaming contexts